diff --git a/kbdd_layout/kbdd_layout.old b/kbdd_layout/kbdd_layout
index 96fd4b61f7cd..19f6afa42874 100755
--- a/kbdd_layout/kbdd_layout.old
+++ b/kbdd_layout/kbdd_layout
@@ -16,11 +16,6 @@
 #	You should have received a copy of the GNU General Public License
 #	along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-# In case user is restarting block after making layout changes
-# e.g. via setxkbmap, reload kbdd
-killall kbdd 2>/dev/null
-kbdd >/dev/null || exit 1
-
 # Get initial state of layout
 N=$( dbus-send --print-reply=literal --dest=ru.gentoo.KbddService\
     /ru/gentoo/KbddService ru.gentoo.kbdd.getCurrentLayout 2>/dev/null |\
