--- ranger/ext/img_display.py.old	2017-08-28 15:57:15.651869415 +0300
+++ ranger/ext/img_display.py	2017-08-28 15:59:41.258670713 +0300
@@ -106,6 +106,8 @@
             start_y, width, height))
         self.process.stdin.flush()
         self.process.stdout.readline()
+        self.quit()
+        self.is_initialized = False
 
     def clear(self, start_x, start_y, width, height):
         if not self.is_initialized or self.process.poll() is not None:
