diff --git a/renpy/character.py.old b/renpy/character.py
index 33e839579269..cb1bec476919 100644
--- a/renpy/character.py.old
+++ b/renpy/character.py
@@ -700,7 +700,7 @@ class ADVCharacter(object):
             **self.show_args)
 
     # This is called after the last interaction is done.
-    def do_done(self, who, what):
+    def do_done(self, who, what, who_args=None):
         self.add_history("adv", who, what)
 
     # This is called when an extend occurs, before the usual add/show
@@ -878,7 +878,7 @@ class ADVCharacter(object):
 
             # Indicate that we're done.
             if _call_done:
-                self.do_done(who, what)
+                self.do_done(who, what, self.who_args)
 
                 # Finally, log this line of dialogue.
                 if who and isinstance(who, (str, unicode)):
